### ISSUE TITLE
Adding AIC/BIC information

### ISSUE THREAD

**Issue by matzhaugen** [sebp/scikit-survival#340]
**Is your feature request related to a problem? Please describe.**
I'm not sure if this is 100% relevant to penalized cox regressions, but would it be useful to ad AIC and BIC metrics if cross-validation is inappropriate, e.g. due to data sparsity or for the reason of replicating other work?
In the same vein, a forward and backward selection method could also be employed (but that would probably go in another feature request). 

**Describe the solution you'd like**
Add an AIC and BIC metric to the existing set of metrics.

**Describe alternatives you've considered**
As an alternative one could also compute the AIC/BIC oneself outside out this codebase.

**References and existing implementations**
An existing implementation of this can be found in the sciki learn package [here](https://github.com/scikit-learn/scikit-learn/blob/a7cd0ca44d0af64bc22a7213371e3c11d94dbbe8/sklearn/linear_model/_least_angle.py#L2305-L2309)

One reference paper that employs this with backward selection is here 
https://doi.org/10.1016/j.jtcvs.2017.11.095

Again, not sure if the other criteria already implemented are perhaps more appropriate. If so, let me know.

**Comment by sebp** [comment]
I agree that having AIC/BIC for CoxnetSurvivalAnalysis would be nice. If you want to work on it, please feel free to open a pull request.

Regarding, forward/backward selection, sklearn's [SelectKBest](https://scikit-learn.org/stable/modules/generated/sklearn.feature_selection.SelectKBest.html#sklearn.feature_selection.SelectKBest) could be helpful in that regard.
